Tensile/TensileCreateLibrary.py

```diff
@@ -484,91 +486,101 @@ def prepAsm(
     os.chmod(assemblerFileName, 0o777)
 
 
-################################################################################
-def buildKernelSourceAndHeaderFiles(results, outputPath):
-    """
-    Logs errors and writes appropriate info to kernelSourceFile and kernelHeaderFile.
+ProcessedKernelLookup = Dict[str, Any]
 
-    Arguments:
-      results:              list of (err, src, header, kernelName, filename)
-      outputPath:           path to source directory
-      kernelsWithBuildErrs: Dictionary to be updated with kernels that have errors
-      kernelSourceFile:     File to write source data to
-      kernelHeaderFile:     File to write header data to
+
+def collectFilesToWrite(
+    results: List[ProcessedKernelResult],
+    outputPath: Path,
+    lazyLoading: bool,
+    mergeFiles: bool,
+    numMergedFiles: int,
+) -> Tuple[ProcessedKernelLookup, ProcessedKernelLookup]:
+    """Collects and organizes kernel files to be written based on the provided results.
+
+    Args:
+        results: A list of processed kernel results, each containing
+            error status, source code, header, kernel name, and filename.
+        outputPath: The path where the output files should be written.
+        lazyLoading: Flag indicating whether lazy loading is enabled.
+        mergeFiles: Flag indicating whether kernel files should be merged.
+        numMergedFiles: The number of files to merge into if merging is enabled.
 
     Returns:
-      sourceFilenames:      Array containing source kernel filenames
+        A tuple containing:
+            - A dictionary mapping file paths to kernel data:
+                (error status, source code, header code, kernel name).
+            - A dictionary mapping kernel names to an error status.
     """
+    pathJoin = lambda x: os.path.join(os.path.normcase(outputPath), x)
 
-    # Find kernels to write
-    kernelsToWrite = []
-    kernelsWithBuildErrs: Dict[str, int] = {}
     filesToWrite = collections.defaultdict(list)
+    kernelsWithBuildErrs = collections.defaultdict(int)
     validKernelCount = 0
-    for err, src, header, kernelName, filename in results:
 
-        # Keep track of kernels with errors
+    for err, src, header, kernelName, filename in results:
         if err:
             kernelsWithBuildErrs[kernelName] = err
```

Comment:
Seems a little awkward to leave this logic in this function. Would be nice to separate into another function such as:

```
def kernelsWithBuildErrors(results: List[ProcessedKernelResult]):
  return {kernelName: err for err, *, *,  kernelName, * in results if err}
```

At this point, do we even need a function?